jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -489,24 +505,26 @@ private HttpTokens.Token next(ByteBuffer buffer)
      */
     private boolean quickStart(ByteBuffer buffer)
     {
+        int position = buffer.position();
         if (_requestHandler != null)
         {
             _method = HttpMethod.lookAheadGet(buffer);
             if (_method != null)
             {
                 _methodString = _method.asString();
-                buffer.position(buffer.position() + _methodString.length() + 1);
+                buffer.position(position + _methodString.length() + 1);
```

Comment:
This seems unsafe.
If I get a buffer with just `POST` in it, I think here we set the position to `0 + 4 + 1 == 5`, which is beyond `limit()` and throws.
If that is not the case, deserves a comment about why not.

Reply:
OK, will look.... but this has not been changed by this PR, I only added a variable.